tests/test_setup_new_composite_areas_ground_elevation.py

```diff
@@ -0,0 +1,89 @@
+)
+
+EXAMPLEDIR = Path().absolute() / "examples" / "data" / "setup_new_composite_area"
+DATADIR = Path().absolute() / "hydromt_fiat" / "data"
+DATADIRDEM = Path(
+    "P:/11207949-dhs-phaseii-floodadapt/FloodAdapt/Test_data/Database_env_fix/static/dem"
+)
+
+_cases = {
+    "setup_new_composite_area_datum": {
+        "dir": "test_read",
+        "new_root": EXAMPLEDIR / "test_setup_new_composite_area_datum",
+        "composite_areas": DATADIR / "new_composite_areas" / "new_development_area_test.gpkg",
+        "type": "datum",
+        "path_ref": None,
+        "attr_ref": None,
+        "ground_elevation_file": DATADIRDEM 
+        / "charleston_14m.tif",
+    },
+    "setup_new_composite_area_geom": {
+        "dir": "test_read",
+        "new_root": EXAMPLEDIR / "test_setup_new_composite_area_geom",
+        "composite_areas": DATADIR / "new_composite_areas" / "new_development_area_test.gpkg",
+        "type": "geom",
+        "path_ref": DATADIR / "new_composite_areas" /  "reference_groundHeight_test.shp",
+        "attr_ref": "bfe",
+        "ground_elevation_file": DATADIRDEM 
+        / "charleston_14m.tif",
+    },
+    "setup_new_composite_area_elevation": {
+        "dir": "test_read",
+        "new_root": EXAMPLEDIR / "test_setup_new_composite_area_elevation",
+        "composite_areas": DATADIR / "new_composite_areas" / "new_development_area_test.gpkg",
+        "type": "datum",
+        "path_ref": None,
+        "attr_ref": None,
+        "ground_elevation_file": DATADIRDEM 
+        / "charleston_14m.tif",
+    },
+}
+
+@pytest.mark.parametrize("case", list(_cases.keys()))
+def test_setup_new_composite_areas_ground_elevation(case):
+    # Read model in examples folder.
+    root = EXAMPLEDIR.joinpath(_cases[case]["dir"])
+    logger = setuplog("hydromt_fiat", log_level=10)
+
+    fm = FiatModel(root=root, mode="r", logger=logger)
+    fm.read()
+
+    # store original exposure
+    exposure_original = fm.exposure.exposure_db
+
+    fm.exposure.setup_new_composite_areas(
+        percent_growth=10,
+        geom_file=str(_cases[case]["composite_areas"]),
+        ground_floor_height=2,
+        damage_types=["Structure", "Content"],
+        vulnerability=fm.vulnerability,
+        elevation_reference=_cases[case]["type"],
+        path_ref=_cases[case]["path_ref"],
+        attr_ref=_cases[case]["attr_ref"],
+        ground_elevation=_cases[case]["ground_elevation_file"],
+    )
+
+    if _cases[case]["new_root"].exists():
+        shutil.rmtree(_cases[case]["new_root"])
+
+    fm.set_root(_cases[case]["new_root"])
+    fm.write()
+
+    # read modified exposure
+    exposure_modified = pd.read_csv(
+        _cases[case]["new_root"] / "exposure" / "exposure.csv"
+    )
+
+    exposure_modified = fm.exposure.exposure_db
+
+    # check if the new development area was added
+    assert len(exposure_modified) > len(exposure_original), 'The aggregated areas were not added'
```

Comment:
Maybe you can also add here the check that the Ground Elevation column is present in the exposure.csv. You can do that with for example

`assert "Ground Elevation" in fm.exposure.exposure_db.columns`

and

`assert fm.exposure.exposure_db["Ground Elevation"].isna().sum() == 0`